src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -1,654 +1,650 @@
-            }
 
-            if (worker.isWorkerAtSiteWithMove(currentPathTarget, PATH_CLOSE)
-                  || ((AbstractBuildingGuards) building).getTask().equals(AbstractBuildingGuards.Task.FOLLOW)
-                  || ticksAtSamePos >= SWITCH_TARGET_AFTER_TICKS)
+            if (worker.getDistance(target) > getAttackRange())
             {
-                return getNextPatrollingTarget((AbstractBuildingGuards) building);
+                worker.isWorkerAtSiteWithMove(target.getPosition(), getAttackRange());
+                return GUARD_ATTACK_PHYSICAL;
             }
-        }
 
-        return GUARD_SEARCH_TARGET;
-    }
-
-    /**
-     * Retrieves the next patrolling target from the guard tower.
-     *
-     * @param building his building.
-     * @return the next state to go to.
-     */
-    private AIState getNextPatrollingTarget(final AbstractBuildingGuards building)
-    {
-        if (building.shallPatrolManually() && building.getTask().equals(AbstractBuildingGuards.Task.PATROL))
-        {
-            worker.setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.patrolling"));
+            final int swordSlot = InventoryUtils.getFirstSlotOfItemHandlerContainingTool(new InvWrapper(getInventory()),
+              ToolType.SWORD,
+              0,
+              getOwnBuilding().getMaxToolLevel());
 
-            final BlockPos pos = building.getNextPatrolTarget(currentPathTarget);
-            if (pos != null)
+            if (swordSlot != -1)
             {
-                currentPathTarget = pos;
-                return GUARD_SEARCH_TARGET;
-            }
-        }
-        else if (building.getTask().equals(AbstractBuildingGuards.Task.GUARD))
-        {
-            worker.setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.guarding"));
+                worker.setHeldItem(EnumHand.MAIN_HAND, swordSlot);
 
-            BlockPos pos = building.getGuardPos();
-            if (pos == null)
-            {
-                pos = building.getLocation();
-            }
-            currentPathTarget = pos;
-            return GUARD_SEARCH_TARGET;
-        }
-        else if (building.getTask().equals(AbstractBuildingGuards.Task.FOLLOW))
-        {
-            worker.setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.following"));
+                worker.faceEntity(target, (float) TURN_AROUND, (float) TURN_AROUND);
+                worker.getLookHelper().setLookPositionWithEntity(target, (float) TURN_AROUND, (float) TURN_AROUND);
 
-            BlockPos pos = building.getPlayerToFollow();
-            if (pos == null || !building.getColony().isCoordInColony(world, pos))
-            {
-                if (pos != null)
+                worker.swingArm(EnumHand.MAIN_HAND);
+                worker.playSound(SoundEvents.ENTITY_PLAYER_ATTACK_SWEEP, (float) BASIC_VOLUME, (float) getRandomPitch());
+
+                double damageToBeDealt = 3;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)